src/python/WMCore/WMExceptions.py

```diff
@@ -22,37 +22,278 @@
 
 """
 WM_JOB_ERROR_CODES
-List of job error codes present in WMCore, some of them coming from CMSSW, and a description
-of the error it represents.
+List of job error codes present in WMCore, some of them coming from CMSSW, and a description of the error it represents.
+
+----------------
+Range(1 - 512)
+    standard ones in Unix and indicate a CMSSW abort that the cmsRun did not catch as exception
+----------------
+Range(7000 - 9000)
+    cmsRun (CMSSW) exit codes. These codes may depend on specific CMSSW version
+    https://github.com/cms-sw/cmssw/blob/CMSSW_5_0_1/FWCore/Utilities/interface/EDMException.h#L26
+----------------
+Range(10000 - 19999)
+    Failures related to the environment setup
+----------------
+Range(50000 - 59999)
+    Failures related executable file
+----------------
+Range(60000 - 69999)
+    Failures related staging-OUT
+----------------
+Range(70000 - 79999)
+    Failures related only for WMAgent. (which does not fit into ranges before)
+----------------
+Range(80000 - 89999)
+    Failures related only for CRAB3. (which does not fit into ranges before)
+----------------
+Range(90000 - 99999)
+    Other problems which does not fit to any range before.
+----------------
+"""
+
+WM_JOB_ERROR_CODES = {
```

Comment:
Let's also get a link to the TWiki in here with a strong admonition that changes must be made in both places.